internal/engine/compiler/impl_amd64_test.go

```diff
@@ -391,6 +391,42 @@ func TestAmd64Compiler_readInstructionAddress(t *testing.T) {
 	})
 }
 
+func TestAmd64Compiler_preventCrossedTargetdRegisters(t *testing.T) {
+	env := newCompilerEnvironment()
+	compiler := env.requireNewCompiler(t, newAmd64Compiler, nil).(*amd64Compiler)
+
+	rv := []*runtimeValueLocation{{register: amd64.RegAX}, {register: amd64.RegCX}, {register: amd64.RegDX}}
```

Comment:
Shoulnd't this be defined inside each case? what if the each case has side effects on this and somehow passes the assertions in a wrong way?